### PR TITLE
Refactor constraint interfaces #LMR-1926

### PR DESCRIPTION
**Pull request by livthomas**

### CODE REVIEW

File: src/app/core/service/translation.service.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class TranslationService {
   public translateDurationUnit(unit: DurationUnit): string {
     return this.i18n(
       {
-        id: 'constraint.duration.unit',
+        id: 'index.ts.duration.unit',
```
Comment (kubedo8) [inline src/app/core/service/translation.service.ts]: Just a refactor typo

File: src/app/shared/pipes/constraint-type-icon-title.pipe.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class ConstraintTypeIconTitlePipe implements PipeTransform {
 
     return this.i18n(
       {
-        id: 'constraint.type.icon.title',
+        id: 'index.ts.type.icon.title',
```
Comment (kubedo8) [inline src/app/shared/pipes/constraint-type-icon-title.pipe.ts]: Another refactor typo

**Comment by kubedo8** [comment]
Looks very good! I see how it simplify our work with constraints.
I will test it in few hours.

**Comment by marvec** [comment]
Did you test the constraints in table? I cannot start entering values for percentage and duration constraint properly. Percentage use wrong formatting and duration always enters 0. Also, kanban-converter.ts seems to contain errors.

**Comment by livthomas** [comment]
I've resolved all the issues you pointed out except for the one in duration data input. The behavior is exactly the same as in our production instance so if we want to change that it will probably be a task for @kubedo8 as he's going to optimize `DurationDataValue` anyway. It is also not possible to enter zero into duration data input at the moment (and it seems it's always been like this).

**Comment by livthomas** [comment]
I've fixed entering numbers to duration data input.

**Comment by livthomas** [comment]
The resizing of data inputs should work now.

**Pull request merged**